tests/apollo/test_skvbc_checkpoints.py

```diff
@@ -123,6 +123,114 @@ async def test_checkpoint_propagation_after_crashed_replicas_comeback(self, bft_
+            bft_network=bft_network,
+            nb_crashing=f - 1,
+            exclude_replicas={expected_next_primary, initial_primary}
+        )
+        self.assertFalse(initial_primary in crashed_replicas or expected_next_primary in crashed_replicas)
+
+        self.assertGreaterEqual(
+            len(bft_network.procs), 2 * f + 2 * c + 1,
+            "Make sure enough replicas are up to allow a successful view change!")
+
+        await skvbc.fill_and_wait_for_checkpoint(
+            initial_nodes=bft_network.all_replicas(without=crashed_replicas),
+            checkpoint_num=1,
+            verify_checkpoint_persistency=False
+        )
+
+        # verify checkpoint creation at the initial primary after f - 1 replicas down
+        checkpoint_init_primary_after = await bft_network.wait_for_checkpoint(replica_id=initial_primary)
+        self.assertEqual(checkpoint_init_primary_after, 1 + checkpoint_init_primary_before)
+
+        await bft_network.wait_for_view(
+            replica_id=initial_primary,
+            expected=lambda v: v == initial_primary,
+            err_msg="Make sure we are in the initial view "
+                    "before crashing the primary."
+        )
+
+        # trigger a view change by crashing the initial primary
+        bft_network.stop_replica(initial_primary)
+        crashed_replicas.add(initial_primary)
+
+        try:
+            with trio.move_on_after(seconds=1):  # seconds
+                await skvbc.send_indefinite_write_requests()
+
+        except trio.TooSlowError:
+            pass
+
+        finally:
+            # wait for view change
+            await bft_network.wait_for_view(
+                replica_id=random.choice(bft_network.all_replicas(without=crashed_replicas)),
+                expected=lambda v: v == expected_next_primary,
+                err_msg="Make sure view change has been triggered."
+            )
+
+        current_primary = initial_primary + 1
+        self.assertEqual(current_primary, expected_next_primary)
+
+        checkpoint_current_primary = await bft_network.wait_for_checkpoint(replica_id=current_primary)
+
+        # verify checkpoint propagation to current primary in the new view
+        self.assertEqual(checkpoint_current_primary, checkpoint_init_primary_after)
+
+        bft_network.start_replicas(crashed_replicas)
+
+        # verify view stabilization among all the stale nodes after they come back up
+        for crashed_replica in crashed_replicas:
+            await bft_network.wait_for_view(
+                replica_id=crashed_replica,
+                expected=lambda v: v == expected_next_primary,
+                err_msg=f"Make sure view change has been triggered for stale replica: {crashed_replica}."
+            )
+
+        # verify checkpoint propagation to all the stale nodes after they come back up
+        for crashed_replica in crashed_replicas:
+            checkpoint_stale = await bft_network.wait_for_checkpoint(
```

Comment:
I'm not sure why this is failing... I'd suggest logging a P1 JIRA bug, and decorating the test with `@unittest.skip("Unstable due to BC-XXXX")`. Then we can merge it into master and you'll investigate the bug separately.

As for the bug investigation, I guess there's no way around digging deeper in the logs and the C++ code related to checkpoints. 
Maybe a good starting point would be to track where the `last_stored_checkpoint` metric gets updated.
My guess is, the metric may not be correctly updated when crashed replicas catch-up with the others via `ReplicaStatusMsg`.

Reply:
Thanks @teoparvanov! Makes sense, will do as you suggested.